src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [, , , initialSeparator] = fromFormat;
+  const [, , , resultingSeparator] = toFormat;
+  const initialArray = fromFormat.slice(0, 3);
+  const resultingArray = toFormat.slice(0, 3);
+  const dateArray = date.split(initialSeparator);
+  const result = [];
+
+  for (const initialElement of initialArray) {
+    for (const resultingElement of resultingArray) {
+      const initialIndex = initialArray.indexOf(initialElement);
+      const resultingIndex = resultingArray.indexOf(resultingElement);
+
+      if (initialElement === resultingElement) {
+        result[resultingIndex] = dateArray[initialIndex];
+        continue;
+      }
+
+      if (initialElement.includes('YY') && resultingElement.includes('YY')) {
+        if (initialElement.length < resultingElement.length) {
+          if (+dateArray[initialIndex] < 30) {
+            result[resultingIndex] = `20${dateArray[initialIndex]}`;
+          } else {
+            result[resultingIndex] = `19${dateArray[initialIndex]}`;
+          }
+        } else {
+          result[resultingIndex] = dateArray[initialIndex].slice(2);
+        }
+      }
+    }
```

Comment:
The nested loops on lines 18-39 result in inefficient code, as they cause unnecessary repetition. The inner loop will repeatedly search for the same elements and indices, which is not required for the task. Instead, you should directly map each element from the `fromFormat` to its corresponding element in `toFormat` using their indices. This can be done in a single loop over `dateArray`.